src/opa.js

```diff
@@ -253,6 +254,59 @@ async function _loadPolicy(policyWasm, memory, customBuiltins) {
   return { policy: wasm, minorVersion: abiMinorVersion };
 }
 
+/**
+ * _loadPolicy can take in either an ArrayBuffer or WebAssembly.Module
+ * as its first argument, a WebAssembly.Memory for the second parameter,
+ * and an object mapping string names to additional builtin functions for
+ * the third parameter.
+ * It will return a Promise, depending on the input type the promise
+ * resolves to both a compiled WebAssembly.Module and its first WebAssembly.Instance
+ * or to the WebAssemblyInstance.
+ * @param {BufferSource | WebAssembly.Module} policyWasm
+ * @param {WebAssembly.Memory} memory
+ * @param {{ [builtinName: string]: Function }} customBuiltins
+ * @returns {Promise<{ policy: WebAssembly.WebAssemblyInstantiatedSource | WebAssembly.Instance, minorVersion: number }>}
+ */
+async function _loadPolicy(policyWasm, memory, customBuiltins) {
+  const env = {};
+
+  const wasm = await WebAssembly.instantiate(
+    policyWasm,
+    _importObject(env, memory, customBuiltins),
+  );
+
+  return _preparePolicy(env, wasm, memory);
+}
+
+/**
+ * _loadPolicySync can take in either an ArrayBuffer or WebAssembly.Module
+ * as its first argument, a WebAssembly.Memory for the second parameter,
+ * and an object mapping string names to additional builtin functions for
+ * the third parameter.
+ * It will return a compiled WebAssembly.Module and its first WebAssembly.Instance.
+ * @param {BufferSource | WebAssembly.Module} policyWasm
+ * @param {WebAssembly.Memory} memory
+ * @param {{ [builtinName: string]: Function }} customBuiltins
+ * @returns {Promise<{ policy: WebAssembly.Instance, minorVersion: number }>}
+ */
+function _loadPolicySync(policyWasm, memory, customBuiltins) {
+  const env = {};
+
+  if (
```

Comment:
Why do we not need this for `_loadPolicy`?

Reply:
`WebAssembly.instantiate` will accept a `BufferSource`, `new WebAssembly.Instance` won't.

Reply:
💡 I've misread those to be the same. They're not. D'oh.

Thanks!